/* 예외처리: 자원 자동 해제시키는 try ~ catch ~ 문장 사용법 */

package step17;

import java.io.FileInputStream;

public class Test14_3 {
  
  static class A {
    
    public void close() throws Exception {
      System.out.println("A.close()");
    }
    
    
  } // class A
  
  static class B implements AutoCloseable {
    
    public void close() throws Exception {
      System.out.println("B.close()");
    } 
    
  } // class B
  
  public static void main(String[] args) {
    System.out.println("step17 Test14_3 실행");
    System.out.println();
    
    int age;

    
    try (
//      A obj1 = new A();
      B obj2 = new B();
    ){
      int age;
      if (age < 19)
        throw new Exception("미성년 입니다.");
    } catch (Exception e) {
      System.out.println("미성년입니다.");
    }
    System.out.println("안녕!");
  }
}


